### PR TITLE
Update Stream documentation

### PR DESCRIPTION
**Pull request by vicb**
### Summary

Change `event.respondWith(...)` for `Response`.

`event.respondWith(...)` is the old Service Worker API that is no more recommended.

Note that the code snippet on this page is correctly using the Module Worker API.

reference: https://developers.cloudflare.com/workers/runtime-apis/fetch/

### Documentation checklist

- [X] The [documentation style guide](https://developers.cloudflare.com/style-guide/) has been adhered to.

### CODE REVIEW

**Review by irvinebroque — APPROVED** [review]
nice one

File: src/content/docs/workers/runtime-apis/streams/index.mdx [hunk 1]
```diff
@@ -15,7 +15,7 @@ The [Streams API](https://developer.mozilla.org/en-US/docs/Web/API/Streams_API)
 
 <DirectoryListing />
 
-Workers do not need to prepare an entire response body before delivering it to `event.respondWith()`. You can use [`TransformStream`](/workers/runtime-apis/streams/transformstream/) to stream a response body after sending the front matter (that is, HTTP status line and headers). This allows you to minimize:
+Workers do not need to prepare an entire response body before returning a `Response`. You can use [`TransformStream`](/workers/runtime-apis/streams/transformstream/) to stream a response body after sending the front matter (that is, HTTP status line and headers). This allows you to minimize:
```
Comment (vicb) [inline src/content/docs/workers/runtime-apis/streams/index.mdx]: @irvinebroque Rereading that paragraph I think "You can use [`TransformStream`](...)" should actually be "You can use a [`ReadableStream`](...)"

Note that `TransformStream` implements `ReadableStream` so the current phrasing is not wrong, just a specialization of the generic case.

I can amend this PR or create a new one.

WDYT?

Comment (irvinebroque) [reply src/content/docs/workers/runtime-apis/streams/index.mdx]: go for it either way

**Review by irvinebroque — APPROVED** [review]

**Comment by vicb** [comment]
@kodster28 could you please help me merge this?

**Review by pedrosousa — APPROVED** [review]

**Pull request merged**